Applications/Spire/Source/KeyBindings/TaskKeysTableView.cpp

```diff
@@ -0,0 +1,414 @@
+  scoped_connection m_source_connection;
+
+  explicit OrderTaskTableModel(
+    std::shared_ptr<OrderTaskArgumentsListModel> source)
+    : m_source(std::move(source)),
+      m_source_connection(m_source->connect_operation_signal(
+        std::bind_front(&OrderTaskTableModel::on_operation, this))) {}
+
+  int get_row_size() const override {
+    return m_source->get_size();
+  }
+
+  int get_column_size() const override {
+    return COLUMN_SIZE;
+  }
+
+  AnyRef at(int row, int column) const override {
+    if(column < 0 || column >= get_column_size()) {
+      throw std::out_of_range("The column is out of range.");
+    }
+    return extract_field(m_source->get(row), static_cast<Column>(column));
+  }
+
+  QValidator::State set(int row, int column, const std::any& value) override {
+    if(column < 0 || column >= get_column_size()) {
+      throw std::out_of_range("The column is out of range.");
+    }
+    auto column_index = static_cast<Column>(column);
+    auto arguments = m_source->get(row);
+    auto previous = to_any(extract_field(arguments, column_index));
+    if(column_index == Column::NAME) {
+      arguments.m_name = std::any_cast<const QString&>(value);
+    } else if(column_index == Column::REGION) {
+      arguments.m_region = std::any_cast<const Region&>(value);
+    } else if(column_index == Column::DESTINATION) {
+      arguments.m_destination = std::any_cast<const Destination&>(value);
+    } else if(column_index == Column::ORDER_TYPE) {
+      arguments.m_order_type = std::any_cast<const OrderType&>(value);
+    } else if(column_index == Column::SIDE) {
+      arguments.m_side = std::any_cast<const Side&>(value);
+    } else if(column_index == Column::QUANTITY) {
+      arguments.m_quantity = std::any_cast<const optional<Quantity>&>(value);
+    } else if(column_index == Column::TIME_IN_FORCE) {
+      arguments.m_time_in_force = std::any_cast<const TimeInForce&>(value);
+    } else if(column_index == Column::TAGS) {
+      arguments.m_additional_tags =
+        std::any_cast<const std::vector<Nexus::Tag>&>(value);
+    } else if(column_index == Column::KEY) {
+      arguments.m_key = std::any_cast<const QKeySequence&>(value);
+    }
+    auto blocker = shared_connection_block(m_source_connection);
+    auto result = m_source->set(row, std::move(arguments));
+    if(result == QValidator::State::Invalid) {
+      return QValidator::State::Invalid;
+    }
+    m_transaction.push(
+      TableModel::UpdateOperation(row, column, previous, value));
+    return result;
+  }
+
+  QValidator::State remove(int row) override {
+    return m_source->remove(row);
+  }
+
+  connection connect_operation_signal(
+      const OperationSignal::slot_type& slot) const override {
+    return m_transaction.connect_operation_signal(slot);
+  }
+
+  void on_operation(const OrderTaskArgumentsListModel::Operation& operation) {
+    visit(operation,
+      [&] (const OrderTaskArgumentsListModel::AddOperation& operation) {
```

Comment:
To properly handle transactions you'll need to add the following cases:

    [&] (const StartTransaction&) {
      m_transaction.start();
    },
    [&] (const EndTransaction&) {
      m_transaction.end();
    },